obspy/io/sac/arrayio.py

```diff
@@ -0,0 +1,637 @@
+        'all' : Do all tests.
+
+    Raises
+    ------
+    SacInvalidContentError
+        Any of the specified tests fail.
+    ValueError
+        'data_hdrs' is specified and data is None, empty array
+        No tests specified.
+
+    """
+    # TODO: move this to util.py and write and use individual test functions,
+    # so that all validity checks are in one place?
+    ALL = ('delta', 'logicals', 'data_hdrs', 'enums', 'reftime', 'reltime')
+
+    if 'all' in tests:
+        tests = ALL
+
+    if not tests:
+        raise ValueError("No validation tests specified.")
+    elif any([(itest not in ALL) for itest in tests]):
+        msg = "Unrecognized validataion test specified"
+        raise ValueError(msg)
+
+    if 'delta' in tests:
+        val = hf[HD.FLOATHDRS.index('delta')]
+        if not (val >= 0.0):
+            msg = "Header 'delta' must be >= 0."
+            raise SacInvalidContentError(msg)
+
+    if 'logicals' in tests:
+        for hdr in ('leven', 'lpspol', 'lovrok', 'lcalda'):
+            val = hi[HD.INTHDRS.index(hdr)]
+            if val not in (0, 1, HD.INULL):
+                msg = "Header '{}' must be {{{}, {}, {}}}."
+                raise SacInvalidContentError(msg.format(hdr, 0, 1, HD.INULL))
+
+    if 'data_hdrs' in tests:
+        try:
+            isMIN = hf[HD.FLOATHDRS.index('depmin')] == data.min()
+            isMAX = hf[HD.FLOATHDRS.index('depmax')] == data.max()
+            isMEN = hf[HD.FLOATHDRS.index('depmen')] == data.mean()
+            if not all([isMIN, isMAX, isMEN]):
+                msg = "Data headers don't match data array."
+                raise SacInvalidContentError(msg)
+        except (AttributeError, ValueError) as e:
+            msg = "Data array is None, empty array, or non-array. " + \
+                  "Cannot check data headers."
+            raise ValueError(msg)
+
+    if 'enums' in tests:
+        for hdr in HD.ACCEP_VALS:
+            val = hi[HD.INTHDRS.index(hdr)]
+            if not is_valid_enum_int(hdr, val, allow_null=True):
+                msg = "Invalid enumerated value, '{}': {}".format(hdr, val)
+                raise SacInvalidContentError(msg)
+
+    if 'reftime' in tests:
+        nzyear = hi[HD.INTHDRS.index('nzyear')]
+        nzjday = hi[HD.INTHDRS.index('nzjday')]
+        nzhour = hi[HD.INTHDRS.index('nzhour')]
+        nzmin = hi[HD.INTHDRS.index('nzmin')]
+        nzsec = hi[HD.INTHDRS.index('nzsec')]
+        nzmsec = hi[HD.INTHDRS.index('nzmsec')]
+
+        # all header reference time fields are set
+        if not all([val != HD.INULL for val in [nzyear, nzjday, nzhour, nzmin, nzsec, nzmsec]]):
+            msg = "Null reference time values detected."
+            raise SacInvalidContentError(msg)
+
+        # reference time fields are reasonable values
+        nzjday_ok = 0 <= nzjday <= 366
+        nzhour_ok = 0 <= nzhour <= 24
+        nzmin_ok = 0 <= nzmin <= 59
+        nzsec_ok = 0 <= nzsec <= 59
```

Comment:
An even simpler way would to just check if the date is actually valid:

```python
try:
    obspy.UTCDateTime(year=nzyear, julday=nzjday, hour=nzhour, minute=nzmin,
                     second=nzsec, microsecond=nzmsec)
except ValueError as e:
    raise SacInvalidContentError("Invalid reference time: %s" % str(e))
```

This also does not allow seconds of 60 but that is something we should deal with in the `UTCDateTime` class.